File: web-face/src/app/core/services/event.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { EventType } from '../models/enums';
import { Event } from '../models/event.model';
import { CrudService } from './crud.service';
import { buildHttpParams } from './http-params.util';

export interface EventSearchParams {
  deviceId?: string;
  userId?: string;
  /** type must be paired with from/to (ISO-8601 instants), matching the backend's /search contract. */
  type?: EventType;
  from?: string;
  to?: string;
}

@Injectable({ providedIn: 'root' })
export class EventService extends CrudService<Event> {
  constructor() {
    super('events');
  }

  search(params: EventSearchParams): Observable<Event[]> {
    return this.http.get<Event[]>(`${this.resourceUrl}/search`, { params: buildHttpParams(params) });
  }
}

File: core/src/main/java/com/csl/lasform/service/EventService.java
package com.csl.lasform.service;

import java.time.Instant;
import java.util.List;

import com.csl.lasform.model.entity.Event;
import com.csl.lasform.model.entity.enums.EventType;

public interface EventService extends CrudService<Event, String> {

    List<Event> createAll(List<Event> entities);

    List<Event> findByDeviceId(String deviceId);

    List<Event> findByUserId(String userId);

    List<Event> findByTypeAndOccurredAtBetween(EventType type, Instant from, Instant to);
}
